packages/controllers/src/snaps/SnapController.ts

```diff
@@ -424,13 +487,43 @@ export class SnapController extends BaseController<
       this._onUnresponsiveSnap,
     );
 
-    this._snapsBeingAdded = new Map();
-    this._maxIdleTime = maxIdleTime;
-    this._maxRequestTime = maxRequestTime;
-    this._idleTimeCheckInterval = idleTimeCheckInterval;
-    this._pollForLastRequestStatus();
-    this._lastRequestMap = new Map();
-    this._rpcHandlerMap = new Map();
+    this.registerMessageHandlers();
+  }
+
+  /**
+   * Constructor helper for registering the controller's messaging system
+   * actions.
+   */
+  private registerMessageHandlers(): void {
+    this.messagingSystem.registerActionHandler(
+      `${controllerName}:addSnap`,
+      (...args) => this.add(...args),
```

Comment:
should we name them the same as the functions, could be confusing that it the action is SnapController:addSnap but really its calling `.add`

Reply:
Yeah, that's fair. I was going back and forth but just straight up aliasing the methods probably makes the most sense.

Reply:
1b76fc1